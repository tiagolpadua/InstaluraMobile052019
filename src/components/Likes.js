import React, { Component } from 'react';
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';

export default class Likes extends Component {
  static carregaIcone(likeada) {
    return likeada
      ? require('../../resources/img/s2-checked.png')
      : require('../../resources/img/s2.png');
  }

  static exibeLikes(likers) {
    if (likers.length <= 0) return null;

    return (
      <Text style={styles.likes}>
        {likers.length} {likers.length > 1 ? 'curtidas' : 'curtida'}
      </Text>
    );
  }

  render() {
    const { foto, likeCallback } = this.props;

    return (
      <View>
        <TouchableOpacity onPress={() => likeCallback(foto.id)}>
          <Image style={styles.botaoDeLike} source={Likes.carregaIcone(foto.likeada)} />
        </TouchableOpacity>

        {Likes.exibeLikes(foto.likers)}
      </View>
    );
  }
}

// c√≥digo anterior omitido
const styles = StyleSheet.create({
  botaoDeLike: {
    marginBottom: 10,
    height: 40,
    width: 40,
  },
  likes: {
    fontWeight: 'bold',
  },
});
