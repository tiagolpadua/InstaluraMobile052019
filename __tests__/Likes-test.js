/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Likes from '../src/components/Likes';
import foto from './foto.json';

it('Deve renderizar corretamente', () => {
  const likes = renderer.create(<Likes foto={foto} likeCallback={jest.fn} />);
  expect(likes).toMatchSnapshot();
});
