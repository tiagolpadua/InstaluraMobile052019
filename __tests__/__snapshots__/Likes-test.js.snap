// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar corretamente 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../resources/img/s2-checked.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": 10,
          "width": 40,
        }
      }
    />
  </View>
</View>
`;
